go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -590,3 +663,128 @@ func getClusterSubnetsExclusion() string {
 	}
 	return exclusion
 }
+
+// AddNodeDNSInfo is called when a NodeDNSInfo object is created, resposible for creating initial data
+// for dnsNames resolved by nodes
+func (efDNS *egressfirewallDNSInformation) AddNodeDNSInfo(nodeDNSInfo *nodednsinfoapi.NodeDNSInfo) error {
+	klog.Infof("Adding NodeDNSInfo %s to the cluster", nodeDNSInfo.Name)
+
+	efDNS.Lock()
+	defer efDNS.Unlock()
+	for dnsName, dnsEntries := range nodeDNSInfo.Spec.DNSEntries {
+		var ipNet []net.IP
+		for _, ip := range dnsEntries.IPAddresses {
+			if efDNS.dnsNameInfo[dnsName].ipNodes[ip] == nil {
+				efDNS.dnsNameInfo[dnsName].ipNodes[ip] = make(map[string]struct{})
+			}
+			efDNS.dnsNameInfo[dnsName].ipNodes[ip][nodeDNSInfo.Name] = struct{}{}
+			ipNet = append(ipNet, net.ParseIP(ip))
+		}
+
+		err := efDNS.dnsNameInfo[dnsName].as.AddIPs(ipNet)
+		if err != nil {
+			return fmt.Errorf("error adding IPs to addressSet for dnsName %s: %v", dnsName, err)
+		}
+
+	}
+	return nil
+}
+
+// UpdateNodeDNSInfo is called when a NodeDNSInfo object is updated, responsible for updating the related internal
+// data
```

Comment:
what internal data? Who updates NodeDNSInfo? What will this do? Needs 10x the docblock.